controller/controller.go

```diff
@@ -207,16 +207,16 @@ func (t *trireme) doHandleCreate(contextID string, policyInfo *policy.PUPolicy,
 // doHandleDelete is the detailed implementation of the delete event.
 func (t *trireme) doHandleDelete(contextID string, runtime *policy.PURuntime) error {
 
+	errS := t.supervisors[t.puTypeToEnforcerType[runtime.PUType()]].Unsupervise(contextID)
+	errE := t.enforcers[t.puTypeToEnforcerType[runtime.PUType()]].Unenforce(contextID)
+
 	t.config.collector.CollectContainerEvent(&collector.ContainerRecord{
 		ContextID: contextID,
 		IPAddress: runtime.IPAddresses(),
 		Tags:      nil,
 		Event:     collector.ContainerDelete,
 	})
 
-	errS := t.supervisors[t.puTypeToEnforcerType[runtime.PUType()]].Unsupervise(contextID)
-	errE := t.enforcers[t.puTypeToEnforcerType[runtime.PUType()]].Unenforce(contextID)
-
 	if errS != nil || errE != nil {
 		return fmt.Errorf("unable to delete context id %s, supervisor %s, enforcer %s", contextID, errS, errE)
```

Comment:
That doesn't make sense to me.

Reply:
Dimitri, This will fix the issue where we might miss some flows when PU exits quickly. 

**Current behavior:**

1. We call `CollectContainerEvent(containerDelete)`, the PU (and its flows) is removed from the cache in enforcer (statscache.go)
2. We now call `Unenforce()`, which will flush all existing flows but this doesn't make sense because the PU is already removed from cache in the previous step

**With this PR:**

1. We call `Unenforce()` which will flush all the stats to the parent.
2. Now we call `CollectContainerEvent(containerDelete)`, PU can now be deleted from the cache without any issue.

>> NOTE: This will also remove the `time.AfterFunc` we added in enforcer, which was causing other issues (https://github.com/aporeto-inc/enforcerd/pull/1037)